### PR TITLE
Backport transactional processing PR (#1222)

### PR DESCRIPTION
**Pull request by KiChjang**

### CODE REVIEW

File: polkadot/xcm/xcm-executor/src/lib.rs [hunk 1]
```diff
@@ -613,79 +660,92 @@ impl<Config: config::Config> XcmExecutor<Config> {
...
-				let to_weigh = self.holding.saturating_take(assets.clone());
-				self.holding.subsume_assets(to_weigh.clone());
-
-				let mut message_to_weigh =
-					vec![ReserveAssetDeposited(to_weigh.into()), ClearOrigin];
-				message_to_weigh.extend(xcm.0.clone().into_iter());
-				let (_, fee) =
-					validate_send::<Config::XcmSender>(dest.clone(), Xcm(message_to_weigh))?;
-				// set aside fee to be charged by XcmSender
-				let parked_fee = self.holding.saturating_take(fee.into());
```
Comment (franciscoaguirre) [inline polkadot/xcm/xcm-executor/src/lib.rs:638]: We need to make sure to keep this fix

Comment (KiChjang) [reply polkadot/xcm/xcm-executor/src/lib.rs:638]: Oops, change got clobbered, adding it back now.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5062051

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5062065

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5062064

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5062066

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5062611

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5063432

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5063567

**Review by franciscoaguirre — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Pull request merged**